src/Api/Serializer/RestSerializer.php

```diff
@@ -248,14 +248,24 @@ function (array $matches) use ($varDefinitions) {
             $relative = substr($relative, 1);
         }
 
-        //Append path to endpoint when leading '//...' present
-        // as uri cannot be properly resolved
-        if ($this->api->isModifiedModel()
-            && strpos($relative, '//') === 0
+        //Append path to endpoint when leading '//...' or
+        // '../' present as uri cannot be properly resolved
+        if (($this->api->isModifiedModel()
+            && strpos($relative, '//') === 0)
         ) {
             return new Uri($this->endpoint . $relative);
         }
 
+        if ($this->api->getServiceName() === 's3'
+            && strpos($relative, '../') !== false
+        ){
+            return new Uri(
+                $this->endpoint
+                . (strpos($relative, '/') !== 0 ? '/' : '')
```

Comment:
So this is just making relative start with a slash if it doesn't?  I might refactor this to say 
```
if ($relative && $relative[0] != '/') {
     $relative = '/' . $relative;
}
 return new Uri($this->endpoint . $relative);
```

if you think that looks better.

Reply:
Also, could we just move this up to the if statement above and short-circuit this there?